concrete-core/src/specification/engines/cleartext_conversion.rs

```diff
@@ -0,0 +1,35 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::CleartextEntity;
+
+engine_error! {
+    CleartextConversionError for CleartextConversionEngine @
+}
+
+/// A trait for engines converting cleartexts.
+///
+/// # Semantics
+///
+/// This [pure](super#operation-semantics) operation generates a cleartext containing the
+/// conversion of the `input` cleartext to a different representation.
```

Comment:
This comment is a bit unclear. Maybe an example could help.

Reply:
A simple example is the conversion from a CPU representation to a GPU representation. The conversion then does an allocation on the GPU and a copy from the CPU to the GPU. I guess we could want to do u32 to u64 conversions but maybe it's a stupid idea :sweat_smile:

Reply:
Good comment ! 

As explained by Agnes, this is meant to allow conversion between two object of the same kind. For instance this could allow to convert between two (different) LWE ciphertext entities. For example one on the gpu and the other on the cpu. In this case, converting would mean moving the data from cpu to gpu memory.

This being said, you are right that this notion representation is not very clear. Actually, we have a meeting to discuss whether we keep it as it currently exist, or whether we drop it. I'll try to make things clearer once this question is settled.

Reply:
Discussing this further with @ilaria-zama and @jborfila , we decided to add `(for instance from cpu to gpu memory)` to make it clear what representation means in this context.